src/toil/wdl/versions/v1.py

```diff
@@ -33,4 +48,736 @@ def analyze(self):
+
+        if fn == 'stdout':
+            return '_toil_wdl_internal__stdout_file'
+        elif fn == 'stderr':
+            return '_toil_wdl_internal__stderr_file'
+        elif fn in ('range', 'zip'):
+            # replace python built-in functions
+            return f'wdl_{fn}'
+
+        call = f'{fn}({params}'
+
+        # append necessary params for i/o functions
+        if fn == 'glob':
+            return call + ', tempDir)'
+        elif fn == 'size':
+            return call + (params + ', ' if params else '') + 'fileStore=fileStore)'
+        elif fn in ('write_lines', 'write_tsv', 'write_json', 'write_map'):
+            return call + ', temp_dir=tempDir, file_store=fileStore)'
+        else:
+            return call + ')'
+
+    def visitArray_literal(self, ctx):
+        """
+        Pattern: LBRACK (expr (COMMA expr)*)* RBRACK
+        """
+        return f"[{', '.join(self.visitExpr(expr) for expr in ctx.expr())}]"
+
+    def visitPair_literal(self, ctx):
+        """
+        Pattern: LPAREN expr COMMA expr RPAREN
+        """
+        return f"({self.visitExpr(ctx.expr(0))}, {self.visitExpr(ctx.expr(1))})"
+
+    def visitIfthenelse(self, ctx):
+        """
+        Ternary expression.
+        Pattern: IF expr THEN expr ELSE expr
+        """
+        if_true = self.visitExpr(ctx.expr(0))
+        condition = self.visitExpr(ctx.expr(1))
+        if_false = self.visitExpr(ctx.expr(2))
+
+        return f'({condition} if {if_true} else {if_false})'
+
+    def visitExpression_group(self, ctx):
+        """
+        Pattern: LPAREN expr RPAREN
+        """
+        return f'({self.visitExpr(ctx.expr())})'
+
+    def visitAt(self, ctx):
+        """
+        Array or map lookup.
+        Pattern: expr_core LBRACK expr RBRACK
+        """
+        expr_core = self._visitExpr_core(ctx.expr_core())
+        expr = self.visitExpr(ctx.expr())
+
+        # parenthesis must be removed because 'i[0]' works, but '(i)[0]' does not
+        if expr_core[0] == '(' and expr_core[-1] == ')':
+            expr_core = expr_core[1:-1]
+
+        return f'{expr_core}[{expr}]'
+
+    def visitGet_name(self, ctx):
+        """
+        Member access.
+        Pattern: expr_core DOT Identifier
+        """
+        expr_core = self._visitExpr_core(ctx.expr_core())
+        identifier = ctx.Identifier().getText()
+
+        if identifier in ('left', 'right'):
+            # hack-y way to make sure pair.left and pair.right are parsed correctly.
+            return f'({expr_core}.{identifier})'
+
+        return f'({expr_core}_{identifier})'
+
+    def visitNegate(self, ctx):
+        """
+        Pattern: NOT expr
+        """
+        return f'(not {self.visitExpr(ctx.expr())})'
+
+    def visitUnarysigned(self, ctx):
+        """
+        Pattern: (PLUS | MINUS) expr
+        """
+        plus: bool = ctx.PLUS() is not None
+        expr = self.visitExpr(ctx.expr())
+
+        if plus:
+            return f'(+{expr})'
+        return f'(-{expr})'
```

Comment:
How come these are down here and not with the other operator code for e.g. + and -?

Reply:
This is defined like that in the [grammar file](https://github.com/openwdl/wdl/blob/main/versions/1.0/parsers/antlr4/WdlV1Parser.g4#L132). I suspect that it has something to do with the order of operations, but I could be wrong.